cdc/sink/ddlsink/mq/ddlproducer/pulsar_ddl_producer.go

```diff
@@ -134,34 +135,33 @@ func NewPulsarProducer(
 // newProducer creates a pulsar producer
 // One topic is used by one producer
 func newProducer(
-	pConfig *pulsarConfig.Config,
+	pConfig *config.PulsarConfig,
 	client pulsar.Client,
 	topicName string,
 ) (pulsar.Producer, error) {
 	po := pulsar.ProducerOptions{
 		Topic: topicName,
 	}
-	if pConfig.BatchingMaxMessages > 0 {
-		po.BatchingMaxMessages = pConfig.BatchingMaxMessages
+	if pConfig.BatchingMaxMessages != nil {
+		po.BatchingMaxMessages = *pConfig.BatchingMaxMessages
 	}
-	if pConfig.BatchingMaxPublishDelay > 0 {
-		po.BatchingMaxPublishDelay = pConfig.BatchingMaxPublishDelay
+	if pConfig.BatchingMaxPublishDelay != nil {
+		po.BatchingMaxPublishDelay = pConfig.BatchingMaxPublishDelay.Duration()
 	}
-	if pConfig.CompressionType > 0 {
-		po.CompressionType = pConfig.CompressionType
+	if pConfig.CompressionType != nil {
+		po.CompressionType = pConfig.CompressionType.Value()
 		po.CompressionLevel = pulsar.Default
 	}
-	if pConfig.SendTimeout > 0 {
-		po.SendTimeout = pConfig.SendTimeout
+	if pConfig.SendTimeout != nil {
+		po.SendTimeout = pConfig.SendTimeout.Duration()
 	}
 
 	producer, err := client.CreateProducer(po)
 	if err != nil {
 		return nil, err
 	}
 
-	log.Info("create pulsar producer success",
-		zap.String("topic:", topicName))
+	log.Info("new pulsar producer ok", zap.String("topic:", topicName))
```

Comment:
I would prefer the previous log content.

Reply:
ok